skimage/io/tests/test_multi_image.py

```diff
@@ -24,6 +26,58 @@ def imgs():
 
     reset_plugins()
 
+
+@pytest.mark.parametrize(
+    "data_filename", ["data/multipage_rgb.tif", "data/no_time_for_that_tiny.gif"]
+)
+def test_debug(data_filename):
+    from skimage.data._fetchers import file_hash, data_dir
+
+    # path_legacy = os.path.abspath(os.path.join(_LEGACY_DATA_DIR, "..", data_filename))
+    path_fetch = testing.fetch(data_filename)
+    path_cache = os.path.abspath(os.path.join(data_dir, "..", data_filename))
+
+    # assert os.path.exists(path_legacy)
+    assert os.path.exists(path_fetch)
+    # assert file_hash(path_legacy) == file_hash(path_fetch)
+
+    assert os.path.exists(path_cache)
+    assert file_hash(path_fetch) == file_hash(path_cache)
+
+
+def test_debug2():
+    from skimage.data._fetchers import data_dir
+
+    # paths_legacy = [
+    #     os.path.abspath(os.path.join(_LEGACY_DATA_DIR, "..", 'data/multipage_rgb.tif')),
+    #     os.path.abspath(
+    #         os.path.join(_LEGACY_DATA_DIR, "..", 'data/no_time_for_that_tiny.gif')
+    #     ),
+    # ]
+    paths_fetch = [
+        testing.fetch('data/multipage_rgb.tif'),
+        testing.fetch('data/no_time_for_that_tiny.gif')
+    ]
+    paths_cache = [
+        os.path.abspath(os.path.join(data_dir, "..", 'data/multipage_rgb.tif')),
+        os.path.abspath(os.path.join(data_dir, "..", 'data/no_time_for_that_tiny.gif')),
+    ]
+
+    # img_legacy = MultiImage(os.pathsep.join(paths_legacy))
+    img_fetch = MultiImage(os.pathsep.join(paths_fetch))
+
+    # assert len(img_legacy) == 2
+    assert len(img_fetch) == 2
+    # np.testing.assert_array_equal(img_legacy[0], img_fetch[0])
+    # np.testing.assert_array_equal(img_legacy[1], img_fetch[1])
+
+    img_cache = MultiImage(os.pathsep.join(paths_cache))
+
+    assert len(img_cache) == 2
+    np.testing.assert_array_equal(img_fetch[0], img_cache[0])
```

Comment:
Getting closer. On [macos-cp3.11 fails on this line](https://github.com/scikit-image/scikit-image/actions/runs/4679595104/jobs/8289818459?pr=6886#step:6:555) with 
```
E           AssertionError: 
E           Arrays are not equal
E           
E           (shapes (25, 14, 4), (2, 10, 10, 3) mismatch)
E            x: array([[[173, 197, 187, 255],
E                   [181, 198, 198, 255],
E                   [166, 183, 172, 255],...
E            y: array([[[[1.222920e-01, 1.532035e-01, 7.516385e-01],
E                    [1.991089e-01, 7.685891e-01, 5.854871e-01],
E                    [1.314398e-01, 3.879300e-01, 9.433194e-01],...
```
Still no idea why.  [`test_debug`](https://github.com/scikit-image/scikit-image/pull/6886/files#diff-53904f4caa90b125ffd8d09ab54bf570df21aacceedce704f3b671d1d2d3c8d2R45) passes, so the actual hash of the fetched and cached file are the same?

Reply:
> Curious, `multipage_rgb.tif` is still included in [skimage/data](https://github.com/scikit-image/scikit-image/tree/1a2e55891c20cb70dfa04a4dbc731a287ad957af/skimage/data) but not in [skimage/data/meson.build](https://github.com/scikit-image/scikit-image/blob/1a2e55891c20cb70dfa04a4dbc731a287ad957af/skimage/data/meson.build). Is that by intention?
> 
> Tough not sure if this is the source of the bug. I can't reproduce this locally and don't have a Mac available...

My guess is that we only included data files in the wheel that are (a) in the repo and (b) used by `skimage.data`. But, frankly, it may just as well have been an oversight.

Reply:
Ha, I think I'm close. The failing test uses two paths one of which now points to the legacy directory. My guess is that the new path is sorted differently in https://github.com/scikit-image/scikit-image/blob/ceaa87d7058c5e8970c52e349a7862c1a3fcea8c/skimage/io/collection.py#L202-L207

for macos-cp3.11 which leads to the arrays in MultiImage having a different order. The mismatching shape in https://github.com/scikit-image/scikit-image/actions/runs/4679595104/jobs/8289818459#step:6:574 indicates such... 

Continuing tomorrow. :sleeping:

Reply:
Confirmed. The paths to were the scikit-image is installed are different, therefore it sorts the paths differently on e.g. [macos-cp3.10](https://github.com/scikit-image/scikit-image/actions/runs/4683653567/jobs/8298971563#step:6:656):
```python
sorted_fetch = [
    '/Users/runner/Library/Caches/scikit-image/main/data/multipage_rgb.tif', 
    '/Users/runner/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/site-packages/skimage/data/no_time_for_that_tiny.gif'
]
```
on [macos-cp3.11](https://github.com/scikit-image/scikit-image/actions/runs/4683653567/jobs/8298971713#step:6:564)
```python
# macos-cp3.11
sorted_fetch = [
    '/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/skimage/data/no_time_for_that_tiny.gif',
    '/Users/runner/Library/Caches/scikit-image/main/data/multipage_rgb.tif'
]
```
I'm not sure why the install location is different or which tool causes it, though I think it's safe to say that the [tests](https://github.com/scikit-image/scikit-image/blob/ceaa87d7058c5e8970c52e349a7862c1a3fcea8c/skimage/io/tests/test_multi_image.py#L13-L23) shouldn't rely rely on the paths being sorted by chance the correct way.